urwid/main_loop.py

```diff
@@ -1324,7 +1324,10 @@ def run(self):
         self._loop.set_exception_handler(self._exception_handler)
         self._loop.run_forever()
         if self._exc_info:
-            raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
+            if PYTHON3:
+                raise self._exc_info
+            else:
+                raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
```

Comment:
Do you have this `if` backwards?  `with_traceback` only exists in Python 3, but this code will only call it for Python 2.

----

I don't understand what the problem was with the original solution from commit 6c719136:

```
raise self._exc_info[1], None, self._exc_info[2]
```

That's "raise value with traceback" in Python 2, which 2to3 will translate to:

```
raise self._exc_info[1].with_traceback(self._exc_info[2])
```

Which reassigns the exception's traceback (a no-op, since it's the same traceback) and throws it as normal.

The followup commit says "sometimes `sys.exc_info()` returns all Nones", but that sets off all kinds of alarm bells for me and seems like something worth investigating rather than hacking around.